### DISCUSSION TITLE
Disable ssl on Mysql Connection

### DISCUSSION THREAD

**Discussion by rrautroku** [apache/airflow#22801] [Q&A]
I am trying to pass `ssl_mode='DISABLED'` to my mysql connection [here](https://github.com/apache/airflow/blob/main/airflow/providers/mysql/hooks/mysql.py#L166). I could not find a way to it. Is there a way this can be done? Additional parameters can be passed through extras in mysql connection in airflow. Disabling ssl is not doable through there.

**Comment by potiuk** [comment]
Look at the way how SQLAlchemy connections works: 

* https://airflow.apache.org/docs/apache-airflow/stable/howto/set-up-database.html?highlight=database#setting-up-a-mysql-database
* https://docs.sqlalchemy.org/en/13/dialects/mysql.html
* https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#sql-alchemy-connect-args
* https://docs.sqlalchemy.org/en/13/core/engines.html#sqlalchemy.create_engine.params.connect_args


If you find a way, it would be great if you post it here so that others can find it too.

**Reply by rrautroku** [reply]
I was able to solve the problem by downgrading to mysqlclient==1.4.6. Could not come up with a way to disable ssl unless I maintain my own version of airflow mysql hook.
